### DISCUSSION TITLE
Rustdesk WAN connection

### DISCUSSION THREAD

**Discussion by guet36** [rustdesk/rustdesk#7885] [General]
hi everybody, 
I have a question please
my server works well in LAN
I have an Ubuntu server, a “technician” machine to take control (on Windows), and a windows client machine
when I move the client machine on the wan (connected to the wifi access point created by my smartphone), the rustdesk client displays "ready", but when I want to take control it does not work.
I created nat rules on my box on ports 21115 to 21119 (hence the “ready” mention) and redirected to the server IP (not that of the “technician” machine).
should I host a DNS server role on my ubuntu server?
Do you have a tutorial please?
THANKS

**Comment by xlionjuan** [comment]
Please provide your `docker-compose.yml` if you're using Docker, or systemctl script at `/usr/lib/systemd/system`

Don't forget hide your address when posting `HBBS`'s args.

**Reply by guet36** [reply]
tks for your quick answer.
sorry but i have no docker-compose and no systemctl script at /usr/lib/systemd/system
i have deployed rustdesk by download and unzip like explain in 
https://www.malekal.com/comment-installer-son-serveur-de-relai-rustdesk/
tks a lot

**Reply by xlionjuan** [reply]
IMO I'm very recommended to deploy RustDesk Server though Docker, it is more convenient, and easy to migrate if you want in the future.

**Comment by guet36** [comment]
ok tks, 
i have build this morning a new server (ubuntu server 24.04 LTS) with ubuntu desktop 
i have try many times to install rustdesk server with docker but no one give me satisfaction.
please, could you help me to find a tutorial to install it as you recommended ?
tks
guet

**Reply by xlionjuan** [reply]
Follow this Synology guide, and use this `docker-compose.yml`

But put it at your home directory 

https://rustdesk.com/docs/en/self-host/rustdesk-server-oss/synology/dsm-7/

How to install docker on Ubuntu:
Two methods:
1. Snap
```
sudo snap install docker 
```

2. Normal apt:
```
sudo apt install docker.io docker-compose 
```

**Comment by guet36** [comment]
tks for quick reply
i have done the installation and it works when all 3 machines are on the same LAN.
but not when client machine is outside

here is my .yml:
version: '3'
services:
  hbbs:
    container_name: hbbs
    image: rustdesk/rustdesk-server:latest
    command: hbbs -r 192.168.x.x:21117 -k _
    volumes:
      - ./data:/root
    network_mode: host
    depends_on:
      - hbbr
    restart: always

  hbbr:
    container_name: hbbr
    image: rustdesk/rustdesk-server:latest
    command: hbbr -k _
    volumes:
      - ./data:/root
    network_mode: host
    restart: always

# Because using docker host mode
# Just in case you forgot the ports:
# 21115 TCP For NAT type test
# 21116 TCP TCP hole punching
# 21116 UDP Heartbeat/ID server
# 21117 TCP Relay

**Reply by xlionjuan** [reply]
By the way, to create a code block with markdown, type this
![Screenshot_20240503-191817.png](https://github.com/rustdesk/rustdesk/assets/25243894/9f4b5567-4888-42fc-979b-66e0b0dea554)


And you could add language, so it will have color

```yaml
version: '3'
```

![Screenshot_20240503-192007.png](https://github.com/rustdesk/rustdesk/assets/25243894/7bccb463-f249-46c1-b93b-aa402b87f3d7)

**Reply by xlionjuan** [reply]
@guet36 
Check here, you should make HBBS point to your HBBR server's WAN IP or Domain 

![Screenshot_20240503-192125.png](https://github.com/rustdesk/rustdesk/assets/25243894/cbdc3b22-b05b-4268-b73c-1c681c7a3d38)

**Reply by guet36** [reply]
i have replace lan ip by public ip in .yml
stop / start again docker services
but client machine don't see server.
any dns configuration to set up ?

**Reply by xlionjuan** [reply]
Can't understand what is **"client can't see server"**

Are you sure command is right?

And you don't need to stop service after you modify `docker-compose.yml`

Just run 
```
sudo docker-compose up -d
```

It will auto recreate with new config.

**Reply by guet36** [reply]
I would say that on the RustDesk client software on the client machine, it is not writing « ready ». 
i have change ne name of exe to put configuration inside (public Ip and key)

**Reply by guet36** [reply]
I have missed one thing about my configuration :
I am behind a Livebox (from orange provider).

**Reply by xlionjuan** [reply]
So, it works on WAN, not works on LAN?

If so, check this 

https://rustdesk.com/docs/en/self-host/nat-loopback-issues/

**Reply by guet36** [reply]
No, it will be the opposite 😃
No work on Wan but works on Lan

**Reply by xlionjuan** [reply]
You need port forwarding

**Reply by guet36** [reply]
yes, i have done this conf :
![Capture d'écran 2024-05-04 103646](https://github.com/rustdesk/rustdesk/assets/168647290/72733996-074a-43fe-8842-9afb8cc79f33)
is it correct for you ?
guetitrustdeskserver-1 is my ubuntu server name

**Reply by xlionjuan** [reply]
Try 21116, 21117

https://canyouseeme.org/

**Reply by xlionjuan** [reply]
21118/21119 isn't necessary, because it is for web client, but web client isn't complete at this moment.

**Reply by guet36** [reply]
![21116](https://github.com/rustdesk/rustdesk/assets/168647290/1acf2168-f0f7-47cf-a63c-d8dcf0e103f8)
![21117](https://github.com/rustdesk/rustdesk/assets/168647290/be2da22f-fa4a-4e22-9711-a23cad4d34cd)
so, i will check why only one of 2 works !
tks

**Reply by xlionjuan** [reply]
21116 both need TCP, UDP

**Reply by guet36** [reply]
ports are now open
![21116 ok](https://github.com/rustdesk/rustdesk/assets/168647290/e3e5e7de-a97b-48dd-80a3-8040c6ea82fc)

i have change 21116 NAT with UDP and TCP ports

![21116 tcpudp](https://github.com/rustdesk/rustdesk/assets/168647290/e0350958-9116-417d-ba9c-12aeb99fea68)

but rustdesk client is still 
"not ready"

**Reply by xlionjuan** [reply]
Ensure key is correct, and your server address is correct

**Reply by guet36** [reply]
hi, 
key is correct (because it works on LAN)
about the server address, there is multiple case :
LAN ip adress
WAN ip addess (public ip)
DNS name server
....
here is the result of "NETSTAT" command
i see that there is no ip adress before ports 21116 to 21119. if the issue could be here ?
![netstat](https://github.com/rustdesk/rustdesk/assets/168647290/4d8a907b-ce5b-4a27-8a39-19500b614ce1)

another question, 
could yuou explain what is : 
server adress
server relay
server API
in the rustdesk client configuration 

tks again !

**Reply by xlionjuan** [reply]
See the right side of your command output, it mean there is a program **listening**  that port.

**Reply by xlionjuan** [reply]
Ensure you typed ID Server and Key on the client, don't touch relay server and API server

![Screenshot_20240505-162908.png](https://github.com/rustdesk/rustdesk/assets/25243894/37abbfbd-d680-4794-a588-94c64dd8d980)

**Comment by guet36** [comment]
![IMG_2599](https://github.com/rustdesk/rustdesk/assets/168647290/9d55b18a-f819-4d00-baf7-8b88acb0f17c)
Could you explain difference between the 3 server informations (ID, Relay, API) ?
Where I have to put public ip ? L’an ip ? …
Many thanks for your help !!
Guet

**Comment by dimejo** [comment]
> Could you explain difference between the 3 server informations (ID, Relay, API) ?

Server ID is the public IP or hostname of your server. Relay is only used if your relay server (hbbr process) is not run on the same host as the ID server. API is only needed for the RustDesk Pro.

> L’an ip ? …

LAN IPs should be irrelevant for your setup.
